lib/use-observer.js

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState, useRef } from 'react';
+import Observer from './can-observer';
+
+export default function useObserver(name) {
+  const [ , update ] = useState();
```

Comment:
I wonder if we should also expose a "function decorator", similar to [`React.memo`](https://reactjs.org/blog/2018/10/23/react-v-16-6.html#reactmemo) where they can just make an observed component like:
```
import { observer } from 'ylem';
import Swanky3rdPartyComponent from 'swank-component-library'

export default observer(Swanky3rdPartyComponent)
```

which would just be a shorthand for 
```
export default props => {
   useObserver();
  return <Swanky3rdPartyComponent {...props} />
}
```

Reply:
We certainly could. Is this a pattern we want to encourage? If so, do we want to expose this *instead* of `useObserver`?

Reply:
I'm not sure... but yeah, I think we should, at least offer both. But I am open to being talked out of it.